src/braket/experimental/autoqasm/api/api.py

```diff
@@ -134,10 +158,29 @@ def _wrapper(*args, **kwargs) -> aq_program.Program:
     return _decorator
 
 
+def _validate_subroutine_args(user_config: aq_program.UserConfig) -> None:
+    """Validate decorator arguments to subroutines. Subroutines do not currently support
+    decorator parameters; however, we allow them if they match the arguments provided to
+    main.
```

Comment:
I'm not sure if I like allowing them in specific cases - it works fine for num_qubits, but will the same pattern work for other potential arguments? Is it easier to explain/document if we just disallow them completely? Or, if we just log/print a warning that we are ignoring the arguments on subroutines?

Reply:
I think these questions can't be answered until we have other arguments besides num_qubits, or until we distinguish `main` and `subroutine` functions. But I think we can allow/disallow on a parameter by parameter basis.

This docstring is maybe too specific to the current state of affairs for what it should do in general, so I'll remove the second sentence or convert to an inline comment